sublime-text-merlin.py

```diff
@@ -261,27 +314,57 @@ class MerlinTypeAtCursorCmd(sublime_plugin.WindowCommand):
     """
     Return type information around cursor.
     """
+
     def __init__(self, window):
         self.window = window
 
     def run(self):
         global enclosing
 
         view = self.window.active_view()
+        id = view.id()
+        if id not in enclosing or enclosing[id] == None:
+            enclosing[id] = MerlinTypeEnclosing(view)
+
+        enclosing[id].show_deepen()
+
+class MerlinWidenTypeAtCursorCmd(sublime_plugin.WindowCommand):
+    def __init__(self, window):
+        self.window = window
+
+    def run(self):
+        global enclosing
+
+        view = self.window.active_view()
+        id = view.id()
+        if id not in enclosing or enclosing[id] == None:
+            enclosing[id] = MerlinTypeEnclosing(view)
+
+        enclosing[id].show_widen()
+
+class MerlinShowTypesOutput(sublime_plugin.TextCommand):
+    def run(self, edit, args):
+        sig_text = args["text"]
+        syntax_file = args["syntax"]
+        window = self.view.window()
+
+        output = window.create_output_panel("merlin-types.mli")
+        full_region = sublime.Region(0, output.size())
+        output.replace(edit, full_region, sig_text)
 
-        if enclosing is None:
-            enclosing = MerlinTypeEnclosing(view)
+        output.set_syntax_file(syntax_file)
+        output.sel().clear()
+        window.run_command("show_panel", {"panel": "output.merlin-types.mli"})
 
-        enclosing.show_phantom()
 
 
 class MerlinTypeMenu(sublime_plugin.TextCommand):
     """
     Display type information in context menu
     """
     def run(self, edit):
-        enclosing = MerlinTypeEnclosing(self.view)
-        enclosing.show_menu()
+        enclosing[self.view.id()] = MerlinTypeEnclosing(self.view)
```

Comment:
Not critical for this PR: maybe enclsoing should be a field of the merlin view?

Reply:
perhaps, i'll file that as todo